### ISSUE TITLE
Replacing Body Node throws Exception

### ISSUE THREAD

**Issue by campersau** [AngleSharp/AngleSharp#530]
The exception is thrown [here](https://github.com/AngleSharp/AngleSharp/blob/devel/src/AngleSharp/Dom/Internal/Node.cs#L718).

```
The operation would yield an incorrect node tree.
```

```csharp
        [Test]
        public void ReplaceBodyNode()
        {
            var document = ("<html><body><div>abc</div></body></html>").ToHtmlDocument();
            var clone = document.Body.Clone(true);

            var newdocument = ("").ToHtmlDocument();

            clone = newdocument.Import(clone, true);

            newdocument.Body.Replace(clone);
        }
```

Why is this forbidden?

Webbrowsers support this.

**Comment by FlorianRappl** [comment]
I think something in the algorithm has to be wrong. Need to check this.

Thanks for the report!

**Comment by FlorianRappl** [comment]
Fixed, see PR #532.
